src/domain/authentication/index.types.d.ts

```diff
@@ -0,0 +1,37 @@
+type SessionId = string & { readonly brand: unique symbol }
+type SchemaId = string & { readonly brand: unique symbol }
+
+type AuthenticationError = import("./errors").AuthenticationError
+
+type IdentityPassword = string & { readonly brand: unique symbol }
+
+type KratosUserId = string & { readonly brand: unique symbol }
+type SessionToken = string & { readonly brand: unique symbol }
+
+type Identity = {
+  id: KratosUserId
+  schemaId: SchemaId
+  traits: {
+    phone?: PhoneNumber
```

Comment:
Why is this `optional?`? This is often an indication that we are combining use cases. Under what circumstances is `phone` going to be present and when is it not? Are these distinct code paths that can use 2 different types?